### PR TITLE
Issue #7149: Third approach to prevent flashing HTML when submitting CKEditor forms

### LINKED ISSUE #7149: Prevent flashing HTML when submitting CKEditor forms
## Description of the bug

When editing any content type that uses CKEditor5, when you click 'Save', the editor is destroyed and you can see a flash of plain HTML where the editor used to be.  This looks funky and confuses less-savvy users.

See video which demonstrates the behavior on a clean install:

https://github.com/user-attachments/assets/e828b83d-5524-4ba9-86e5-a9b518b40d9d

## Steps To Reproduce

To reproduce the behavior:
1. Create a clean install of Backdrop (optional)
2. Click to "Create Page"
3. Make sure the editor is set to Basic, and using CKEditor for the Body field.
4. Type anything into the Body field
5. Save, and watch closely, you will see a flash of HTML.

## Actual behavior

See video above.

## Solution!

There's no reason to destroy the editor `(line 123 of core/modules.ckeditor5/js/ckeditor5.js)` if the form is being submitted anyway.

I have a kludge/hack that actually fixes this; will submit a PR shortly.  Basically, I set a variable if I know the form is submitting, then, in ckeditor5.js, I only call `editor.destroy()` if we are NOT submitting the form.  This works in my case, but I don't know if it works in all situations.

## Additional information

Add any other information that could help, such as:
- Backdrop CMS version:  1.34.0
- Web server and its version: Apache 2.4.65
- PHP version: 8.1.34
- Database sever (MySQL or MariaDB?) and its version: MariaDB 10.7.8
- Operating System and its version: Debian
- Browser(s) and their versions: Chrome 148

### PR DESCRIPTION
**Pull request by quicksketch**
Fixes https://github.com/backdrop/backdrop-issues/issues/7149

This is an alternative to my earlier PR https://github.com/backdrop/backdrop/pull/5299 and @swampopus's https://github.com/backdrop/backdrop/pull/5279.

Instead of using a new parameter like my previous PR, this uses the existing "trigger" parameter with the value set to "submit". I think this is okay because previously we were not using any value at all when calling this function, other than "serialize". So introducing a new option to differentiate between "unload" (which is what should have been used before but wasn't in place) and "submit" (new option for when non-AJAX submitting the form), seems safe.

### CODE REVIEW

File: core/modules/ckeditor5/js/ckeditor5.js [hunk 1]
```diff
@@ -123,7 +123,7 @@
       if (trigger !== 'serialize') {
         // If submitting the entire form, skip destroying the editor and let it
         //  be unloaded by the browser leaving the page.
-        if (!windowUnload) {
+        if (trigger === 'submit') {
```
Comment (swampopus) [inline core/modules/ckeditor5/js/ckeditor5.js]: Should be `if trigger !== 'submit'`

**Comment by swampopus** [comment]
@quicksketch --- Sorry-- you got it backwards ;)  The editor should only call destroy if `trigger !== 'submit'`

`editor.destroy()` is what causes the flash of HTML to appear.

File: core/modules/ckeditor5/js/ckeditor5.js (hunk 1 above)
Comment (indigoxela) [reply core/modules/ckeditor5/js/ckeditor5.js]: I belief, the desired logic here would be:
```
if (trigger === 'submit') {
  return;
}
```
Doing nothing, also none of the cleanup jobs below.
And probably not inside the `trigger !== serialize` check.

Comment (swampopus) [reply core/modules/ckeditor5/js/ckeditor5.js]: Unfortunately we can't just return here, because lower in the function we have this:
```js
// Save formatted value after destroying the editor, which can also
// update the element value.
element.value = newData;
```
This is what actually writes the plain HTML back to the hidden textarea, which actually gets saved.  So we really do need to simply skip over the .destroy() call, so it does need to be within the !== 'serialize' if block.

Comment (indigoxela) [reply core/modules/ckeditor5/js/ckeditor5.js]: ... but that should be inside some `trigger === "serialize"` check
serialize gets dispatched before unload ... in theory... Edit: yes, that's dispatched before (core/misc/ajax.js).

Oh dear, so much easier with Tiny... :hand_over_mouth:

Comment (swampopus) [reply core/modules/ckeditor5/js/ckeditor5.js]: lol, true.  Probably the whole detach function here could use some sprucing up-- but for today, I think simply changing it to `trigger !== 'submit'` gets it done.

File: core/modules/ckeditor5/js/ckeditor5.js [hunk 2]
```diff
@@ -121,7 +121,11 @@
 
       // Destroy the instance if fully detaching.
       if (trigger !== 'serialize') {
-        editor.destroy();
+        // If submitting the entire form, skip destroying the editor and let it
+        //  be unloaded by the browser leaving the page.
+        if (trigger === 'submit') {
```
Comment (quicksketch) [inline core/modules/ckeditor5/js/ckeditor5.js]:
```suggestion
        if (trigger !== 'submit') {
```

**Comment by swampopus** [comment]
Perfect 😎

**Comment by laryn** [comment]
Thanks -- merged into `1.x` and `1.34.x`.
